app/graphql/types/interfaces/rules_preload.rb

```diff
@@ -12,8 +12,8 @@ def rules(args = {})
     latest_test_result_batch(args).then do |latest_test_result|
       latest_rule_results_batch(latest_test_result).then do |rule_results|
         rules_for_rule_results_batch(rule_results).then do |rules|
-          initialize_rules_context(rules, rule_results, args)
-          rules
+          initialize_rules_context(rules.compact, rule_results, args)
+          rules.compact
```

Comment:
We never got this error as we never got this far in the code when `rules` contains a nil value, but if you do, you still want to get rid of it or you get `{"message"=>"Cannot return null for non-nullable field Profile.rules"}]` in the response